iron-fit-behavior.html

```diff
@@ -349,37 +355,21 @@
       var positionRect = this.__getNormalizedRect(this.positionTarget);
       var fitRect = this.__getNormalizedRect(this.fitInto);
 
-      var margin = this._fitInfo.margin;
+      var position = this.__getPosition(this._localeHorizontalAlign, this.verticalAlign, rect, positionRect, fitRect);
 
-      // Consider the margin as part of the size for position calculations.
-      var size = {
-        width: rect.width + margin.left + margin.right,
-        height: rect.height + margin.top + margin.bottom
-      };
-
-      var position = this.__getPosition(this._localeHorizontalAlign, this.verticalAlign, size, positionRect, fitRect);
+      var margin = this._fitInfo.margin;
+      var minLeft = margin.left + (position.horizontalAlign === 'left' ? this.horizontalOffset : 0);
+      var minTop = margin.top + (position.verticalAlign === 'top' ? this.verticalOffset : 0);
+      var maxRight = fitRect.right - margin.right - (position.horizontalAlign === 'left' ? 0 : this.horizontalOffset);
+      var maxBottom = fitRect.bottom - margin.bottom - (position.verticalAlign === 'top' ? 0 : this.verticalOffset);
 
       var left = position.left + margin.left;
       var top = position.top + margin.top;
+      var right = Math.min(maxRight, left + rect.width);
+      var bottom = Math.min(maxBottom, top + rect.height);
 
-      // Use original size (without margin).
-      var right = Math.min(fitRect.right - margin.right, left + rect.width);
-      var bottom = Math.min(fitRect.bottom - margin.bottom, top + rect.height);
-
-      var minWidth = this._fitInfo.sizedBy.minWidth;
-      var minHeight = this._fitInfo.sizedBy.minHeight;
-      if (left < margin.left) {
-        left = margin.left;
-        if (right - left < minWidth) {
-          left = right - minWidth;
-        }
-      }
-      if (top < margin.top) {
-        top = margin.top;
-        if (bottom - top < minHeight) {
-          top = bottom - minHeight;
-        }
-      }
+      left = Math.min(Math.max(left, minLeft), right - this._fitInfo.sizedBy.minWidth);
+      top = Math.min(Math.max(top, minTop), bottom - this._fitInfo.sizedBy.minHeight);
```

Comment:
These calls to `max` are doing the same thing as `min` in the assignments to `right` and `bottom` but for `left` and `top`; maybe they should be in the assignments also?